CMSIS/Core/Include/cmsis_iccarm.h

```diff
@@ -150,7 +150,12 @@
 #endif
 
 #ifndef   __RESTRICT
-  #define __RESTRICT            __restrict
+  #if __ICCARM_V8
+    #define __RESTRICT            __restrict
+  #else
+    /* Needs IAR language extensions */
```

Comment:
@TTornblom is there a way to check for IAR language extensions? I guess the compiler creates weird error messages when using `restrict` without these extensions? Any chance to give a better hint to the user?